### DISCUSSION TITLE
Rotate seed distribution

### DISCUSSION THREAD

**Discussion by joshuajaistein** [neperfepx/neper#406] [Q&A] [answered]
Is it possible to ratate the crystal seeds and then orient all of them towards a desired angle measure ? I have depicted it on the image here 
![Explanation 1](https://user-images.githubusercontent.com/55456301/157222798-2815f620-8eb2-488f-aed3-2e48a6952085.png)
?

**Comment by rquey** [comment, marked as answer]
This option does not exist, but you can load a custom (rotated) seed distribution using [-morphooptiini coo](https://neper.info/doc/neper_t.html#cmdoption-morphooptiini).

**Reply by joshuajaistein** [reply]
Sir, can you please quote an example ?

**Reply by rquey** [reply]
Here is an example:

First, generate a tessellation:
```
neper -T -n 1000 -statseed x,y,z
```
This will write a tessellation file but also a seed coordinates file, `n1000-id1.stseed`.

Then, apply changes to the coordinates, e.g. a scaling along x:

```
awk '{print $1*0.2,$2,$3}' n1000-id1.stseed > new_coos
```

Load this file to generate the new tessellation (you may load any other (properly formatted) file):
```
neper -T -n 1000 -morphooptiini "coo:file(new_coos)" -o new
```

Visualize:
```
neper -V new.tess -print img
```
![image](https://user-images.githubusercontent.com/2659376/157263292-c52b393f-3447-4db5-a15b-c2097a9c7305.png)

Most grains are smashed on the xmin surface, as expected from the seed coordinates.

**Reply by joshuajaistein** [reply]
How to create that feature in neper ? Which part of the code should I start working on ?

**Reply by rquey** [reply]
This is in [net_tess_opt_init_sset](https://github.com/rquey/neper/tree/main/src/neper_t/net_tess/net_tess_opt/net_tess_opt_init/net_tess_opt_init_sset).

**Comment by joshuajaistein** [comment]
Sir , is it possible to code this feature, if yes, where/which function
should I start with ?

**Reply by rquey** [reply]
See https://github.com/neperfepx/neper/discussions/406#discussioncomment-2322729.
